### PR TITLE
Catches exception (with notification) while trying to get corrupt duplicated env var

### PR DESCRIPTION
**Pull request by maryamariyan**
Provide non-fatal notification about corrupted environment block:
- Throw and catch intentionally

cc: @danmosemsft @jkotas 
Fixes: #30341

### CODE REVIEW

**Comment by maryamariyan** [comment]
I could add a unit test for this code path if there was a straightforward way to set up a duplicated environment variable in the test.

File: src/System.Runtime.Extensions/src/System/Environment.cs [hunk 1]
```diff
@@ -44,7 +45,15 @@ private static Hashtable ToHashtable(this IEnumerable<KeyValuePair<string, strin
...
             {
-                hashTable.Add(pair.Key, pair.Value);
+                try 
+                {
+                    hashTable.Add(pair.Key, pair.Value);                 
+                }
+                catch (Exception ex)
+                {
+                    // Throw and catch intentionally to provide non-fatal notification about corrupted environment block
+                    Debug.WriteLine(SR.Diagnostic_ThrowingException, ex.ToString());
```
Comment (jkotas) [inline src/System.Runtime.Extensions/src/System/Environment.cs]: I do not think we should be writing out anything here. These `Debug.WriteLines` in the core libraries are always super annoying when you hit them.

Just catching the exception is enough.

File: src/System.Runtime.Extensions/src/System/Environment.cs [hunk 2]
```diff
@@ -44,7 +45,15 @@ private static Hashtable ToHashtable(this IEnumerable<KeyValuePair<string, strin
             Hashtable hashTable = new Hashtable();
             foreach (KeyValuePair<string, string> pair in pairs)
             {
-                hashTable.Add(pair.Key, pair.Value);
+                try 
+                {
+                    hashTable.Add(pair.Key, pair.Value);                 
+                }
+                catch (Exception ex)
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/src/System/Environment.cs]: Why Exception and not InvalidOperationException (or whatever)? We should always catch the most specific type possible.

Comment (maryamariyan) [reply src/System.Runtime.Extensions/src/System/Environment.cs]: Sure. I'll switch to ArgumentException

**Comment by danmosemsft** [comment]
> I could add a unit test for this code path if there was a straightforward way in test set up to create a duplicated environment variable.

I think this would require PInvoke. You would construct a block of memory per the documentation for CreateProcessW, but with doubled environment variables. Then invoke CreateProcessW, and within that process, retrieve Environment.GetEnvironmentVariables. As noted in the original discussion, CreateProcessW does not validate this block (in fact that is what we believe is allowing this corruption).

I am not sure that is worth it unless you feel motivated, and I would not spend more than a couple hours on it if you are.

**Pull request merged**